src/App.js

```diff
@@ -4,25 +4,37 @@ import TodoList from "./TodoList";
 import AddTodoForm from "./AddTodoForm";
 
 
-const useSemiPersistentState = () =>{
+function App() {
 
-    const [todoList, setTodoList] = React.useState(
-        JSON.parse(localStorage.getItem('savedTodoList') || '[]' )
-    );
+    const [todoList, setTodoList] = React.useState([]);
+
+    const [isLoading, setIsLoading] = React.useState(true);
 
 
     React.useEffect(() => {
-        localStorage.setItem('savedTodoList', JSON.stringify(todoList));
-    }, [todoList]);
 
-    return [todoList, setTodoList];
-}
+        new Promise((resolve, reject) =>
+            setTimeout(
+                () => resolve({data: {todoList:
+                            JSON.parse(localStorage.getItem('savedTodoList'))
+                }}),
+                2000
+            ),
+        ).then((result) => {
+            setTodoList(JSON.parse(localStorage.getItem('savedTodoList')))
+            setIsLoading(false);
+        }
+        );
 
+    }, []);
 
 
-function App() {
+    React.useEffect(() => {
+        if(isLoading == false) {
```

Comment:
nit: `===` is preferable to `==` pretty much all the time. Also you can just use `!isLoading` here instead